### ISSUE TITLE
MinkowskiNet not learning anything on ModelNet40 Classification

### ISSUE THREAD

**Issue by Nicny** [StanfordVL/MinkowskiEngine#31]
I have labeled 3D point cloud dataset much similar to ScanNet. I only have 3D points and the intensity value of that point. So I am feeding the intensity as a feature and the 3D points as the coordinates.
Here's the code I am using to run the network:
`
criterion = nn.CrossEntropyLoss()
device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
model = ResNet14(in_channels=1, out_channels=40, D=3).cuda()
# model = vgg16(input_channel=1, num_classes=40, D=3).cuda()
optimizer = Adam(model.parameters(), lr=1e-1)

trainIterator = data.train_cls()
validIterator = data.valid()
print('#parameters', sum([x.nelement() for x in model.parameters()]))
for epoch in range(10):
    model.train()
    i = 0
    start = time.time()
    for batch in trainIterator:
        optimizer.zero_grad()

        coords = batch[0]
        feat = batch[1]
        label = batch[2].to(device)

        sinput = ME.SparseTensor(feat, coords=coords).to(device)
        predictions = model(sinput)
        loss = criterion(predictions.F, label)
        i += 1
        print('Iteration: ', i, ', Loss: ', loss.item())
        loss.backward()
        optimizer.step()
    torch.save(epoch, 'epoch.pth')
    torch.save(model.state_dict(), 'model.pth')`
Here is the data preparation code:
`
class dataset_valid_cls(torch.utils.data.Dataset):
    def __init__(self):
        l1 = glob.glob('/home/zhang/桌面/SparseConvNet-master/data/val_pts/*.pts')
        self.path = l1
        print(len(self.path))
    def __len__(self):
        return len(self.path)
    def __getitem__(self, ind):
        curpath = self.path[ind]
        data = load_file(curpath,0.001)
        return data
def load_file(file_name, voxel_size):
	coords = np.loadtxt(file_name)
	feats = np.ones((coords.shape[0], 1))
	labels = np.loadtxt(file_name[:-3] + 'label').astype('int64')
	discrete_coords, unique_feats = ME.utils.sparse_quantize(
		coords=coords,
		feats=feats,
		labels=None,
		hash_type='ravel',
		set_ignore_label_when_collision=False,
		quantization_size=voxel_size)
	return discrete_coords, unique_feats, labels
def collation_fn(data_labels):
    coords, feats, labels = list(zip(*data_labels))
    coords_batch, feats_batch, labels_batch = [], [], []
    for batch_id, _ in enumerate(coords):
        N = coords[batch_id].shape[0]
        coords_batch.append(
            torch.cat((torch.from_numpy(coords[batch_id]).int(),
                       torch.ones(N, 1).int() * batch_id), 1))
        feats_batch.append(torch.from_numpy(feats[batch_id]))
        labels_batch.append(torch.from_numpy(labels[batch_id]))
    # Concatenate all lists
    coords_batch = torch.cat(coords_batch, 0).int()
    feats_batch = torch.cat(feats_batch, 0).float()
    labels_batch = torch.tensor(labels_batch).long()
    return coords_batch, feats_batch, labels_batch
def train_cls():
	return torch.utils.data.DataLoader(dataset_train_cls(),
									   batch_size=20,
									   collate_fn=collation_fn,
									   num_workers=0,
									   shuffle=True)
`
